components/Themes/src/MDCSemanticColorScheme.h

```diff
@@ -0,0 +1,80 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import "MDCColorScheme.h"
+
+/**
+ A simple color scheme that provides more semantic context for the colors it uses. Unlike
+ @c MDCColorScheme, there are no optional properties, so more consistent theming can be applied.
+ */
+@protocol MDCSemanticColorScheming <MDCColorScheme>
+
+/**
+ Displayed most frequently across your app.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *primaryColor;
+
+/**
+ A tonal variation of primary color.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *primaryColorLightVariant;
```

Comment:
Attaching light and dark to these variants will make it harder to implement darkMode.
If lightMode, statusBar.background = primaryColorLightVariant
else if darkMode, statusBar.background = primaryColorDarkVariant

statusBar.background = primaryColorEmphasisVariant

Naming is hard!

Reply:
My understanding presently is that these properties are semantically safe as light/dark variants regardless of the theme being light or dark, but I also have not seen any literal mappings be well defined in any componentry, so the jury's out on whether that's actually the case...

Reply:
At least for now, this is what our cross-platform spec is expecting as far as symbols.

Reply:
As far as I know, I don't think any of these are intended to be background colors.  Internally we are discussing about how far the "variants" should *vary* from the primary color, but it doesn't seem sufficient to support 4.5:1 contrast ratios.  I doubt the intention is for `.dark` to be background and `.light` to be text color (or *vice versa*)

Reply:
Correct, the dark and light variants are not intended to be used for overlaying text.

Reply:
We'll soon be adding onPrimary and onSecondary to the Color Scheme, which are accessibility-safe colors meant to be used for text and iconography overlaid on top of the primary or secondary color, respectively.

Reply:
My point is that LightVariant / DarkVariant aren't great from a semantic standpoint.
Simplifying to primaryColorVariant is more useful.  (Without light / dark)
statusBar.background = primaryColorVariant // Works in light or dark mode

Reply:
We'll be dropping the light/dark variants in favor of a single variant.